### PR TITLE
chore: update changelog for 10.0.0 release

### PR DESCRIPTION
**Pull request by montezume**
#### Summary

(Draft) update of changelog for 10.0.0

### CODE REVIEW

**Comment by montezume** [comment]
Hi, I could use some feedback if this changelog is sufficient for helping users with migration.

Thanks 😄

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,100 @@
...
+| white       | surface   |
+| blue        | info      |
+| green       | primary   |
+| green-light | primary40 |
+| orange      | warning   |
+| red         | error     |
+
+- `Text`: add `as` prop while deprecating `elementType` ([#631](https://github.com/commercetools/ui-kit/issues/631)) ([4bd6b42](https://github.com/commercetools/ui-kit/commit/4bd6b42))
+
+The prop `elementType` has been renamed to `elementType` for Text components. The `elementType` prop still exists, but is deprecated and will be removed in the next major UI-Kit release. Please migrate your components.
```
Comment (jonnybel) [inline CHANGELOG.md]: Also:
> For `Text.Body` component, the `isInline` prop is also being deprecated. The same result is achieved by setting the `as` prop to `span`.

Comment (montezume) [reply CHANGELOG.md]: I'll (try to remember to ) add this on Monday

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,100 @@
-<a name="9.10.0></a>
+<a name="10.0.0"></a>
+
+# [10.0.0](https://github.com/commercetools/ui-kit/compare/v9.10.0...10.0.0) (2019-06-20)
+
+### Breaking Changes
+
+This release introduces **breaking changes** which may entail some **migration steps**. We'll go through them now:
+
+- `hocs`: `withMouseOverState` and `withMouseDownState` have been dropped. Please use CSS properties instead. ([#881](https://github.com/commercetools/ui-kit/pull/881))
```
Comment (emmenko) [inline CHANGELOG.md]: I would maybe add that if people need those, we recommend using hooks. We can also provide a hook snippet that people can copy-paste.

Redux does something similar and I found it very useful.
https://react-redux.js.org/next/api/hooks#hooks-recipes

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,4 +1,100 @@
...
+
+- `Icons`: make icons themeable (breaking change) ([#726](https://github.com/commercetools/ui-kit/issues/726)) ([9bea3ad](https://github.com/commercetools/ui-kit/commit/9bea3ad))
```
Comment (emmenko) [inline CHANGELOG.md]:
```suggestion
- `Icons`: make icons themeable ([#726](https://github.com/commercetools/ui-kit/issues/726)) ([9bea3ad](https://github.com/commercetools/ui-kit/commit/9bea3ad))
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,4 +1,100 @@
...
+
+The prop `theme` has been renamed to `color` for Icons. Also, the values of this prop have changed. Please see the table below for the mapping between the old and new values.
```
Comment (emmenko) [inline CHANGELOG.md]: It would be good to provide a codemod for this kind of changes.

File: CHANGELOG.md (hunk 1 above)
Comment (emmenko) [inline CHANGELOG.md]:
```suggestion
The prop `elementType` has been renamed to `as` for Text components. The `elementType` prop still exists, but is deprecated and will be removed in the next major UI-Kit release. Please migrate your components.
```

Comment (emmenko) [inline CHANGELOG.md]: > Please migrate your components

Also here, a codemod can easily do this.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,4 +1,100 @@
...
+| --border-color-separator         |            --color-neutral             |
+| --border-radius-input            |       --border-radius-for-input        |
+| --border-radius-tag              |        --border-radius-for-tag         |
+
+#### Other
+
+The custom property `transition-standard` has been changed from having a value of `all 0.2s ease` to `200ms ease`. To continue using this property as below, the following refactor is necessary
+
+```
+transition: all var(--transitionStandard);
```
Comment (emmenko) [inline CHANGELOG.md]: Should we maybe mention that you should try to avoid animating all props?

**Comment by montezume** [comment]
Sure, I can address all that. FYI, I would rather release 10.0.0 sooner than later, as it's distracting to have this open branch open for months. If we need to make more breaking changes later, I'm just as happy to release 11.0.0 as well

File: CHANGELOG.md (hunk 4 above)
Comment (jonnybel) [reply CHANGELOG.md]: We already added a codemod in `10.0.0` which can be used for this: https://github.com/commercetools/ui-kit/blob/10.0.0/transforms/react-component-rename-props.js

**Comment by montezume** [comment]
@emmenko codemods will be done in a different branch, see #926

File: CHANGELOG.md (hunk 2 above)
Comment (montezume) [reply CHANGELOG.md]: I don't think this is necessary since we already made these HOCs internal to the MC.

**Comment by montezume** [comment]
I'm going to merge this, we can have further discussion in the 10.0.0 branch.

**Pull request merged**